tsdb/wal/wal.go

```diff
@@ -723,40 +724,38 @@ func (w *WAL) fsync(f *Segment) error {
 }
 
 // Close flushes all writes and closes active segment.
-func (w *WAL) Close() (err error) {
+func (w *WAL) Close() error {
 	w.mtx.Lock()
 	defer w.mtx.Unlock()
 
 	if w.closed {
 		return errors.New("wal already closed")
 	}
 
-	if w.segment == nil {
-		w.closed = true
-		return nil
-	}
```

Comment:
Removing the `nil` check can cause panic below. We should have this after `<-donec`.